bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -155,10 +157,12 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
     oss << invalidClient << ", sentFromReplicaToNonPrimary=" << sentFromReplicaToNonPrimary;
     onReportAboutInvalidMessage(m, oss.str().c_str());
     delete m;
+    MDC_CID_REMOVE(GL);
     return;
   }
 
   if (readOnly) {
+    MDC_CID_REMOVE(GL);
```

Comment:
Why is this line before the delete call?

Reply:
This is because the `executeReadOnlyRequest` method also inserts the CID on its beginning, so we wouldn't want to add it twice.
The reason why `executeReadOnlyRequest` also inserts the CID is for the for when we invoke this method from another part of the code (currently, there isn't such case though).